runtime/common/MeasureCounts.h

```diff
@@ -230,6 +229,9 @@ class sample_result {
   CountsDictionary
   to_map(const std::string_view registerName = GlobalRegisterName) const;
 
+  ExecutionResult get_result_for_register(
+      const std::string_view registerName = GlobalRegisterName) const;
```

Comment:
@amccaskey @boschmitt This is an _added_ method to the `sample_result` type (user-facing), hence I'd like to check if it's okay.

The rationale is that the `to_map` method only returns the count map; this one returns the full `ExecutionResult`, which may contain the `expectationValue`.

Reply:
Seems a bit weird. At least if the idea behind `sample_result` is to wrap `ExecutionResult` as stated by its documentation. You would be basically breaking the abstraction. @amccaskey might have better insights regarding this.

Reply:
Hi. I changed this back and used Thiens solution which doesnt break the abstraction.